### ISSUE TITLE
`unused-noqa` (`RUF100`) - false negatives and strange behavior with multiple codes

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#15682]
### Description

i assume this is an unintentional bug in the logic that allows you to add additional information to noqa comments:

```py
# noqa: D103 self this function is self-documenting
```

but ruff should enforce a space between the code and any additional comments. currently it doesn't seem to do so. it also seems to allow certain completely invalid codes:

```py
# no error even though these aren't valid codes:
def foo(): ... # noqa: D103a
# noqa: Z
```

interestingly, these are treated as two separate codes:
```py
def foo(): ... # noqa: D103F811
```
i think it should instead be an error and force the user to separate them with a comma like this:

```py
def foo(): ... # noqa: D103,F811
```

https://play.ruff.rs/2b3608e9-f97a-459a-960b-f702827ef4e4

**Comment by MichaReiser** [comment]
Thanks, I think we miss a check for `leading_space` to be `!= 0` here

https://github.com/astral-sh/ruff/blob/c39ca8fe6d3fb77684d2919a58b77cca95671ab7/crates/ruff_linter/src/noqa.rs#L105-L128

**Comment by MichaReiser** [comment]
Can you tell me more about the design decisions you see? 

These could also be multiple fixes:

* `D103F811`: Should not be treated as two rules
* `D103a`: Handle content after a rule code

**Comment by MichaReiser** [comment]
Oh, so the problem linked in #14229 is that we incorrectly parsed text coming after a code as a noqa code. 

That means we have to define a heuristic of what we treat as codes and what as non-code. 

I also just found https://github.com/astral-sh/ruff/pull/12811
